src/primitives/block.h

```diff
@@ -116,6 +116,50 @@ class CBlock : public CBlockHeader
         return block;
     }
 
+    /**
+     * Get the vout index of the segwit commitment in the coinbase transaction of this block.
+     *
+     * Returns -1 if no witness commitment was found.
+     */
+    inline int GetWitnessCommitmentIndex() const
+    {
+        return vtx.empty() ? -1 : GetWitnessCommitmentIndex(*vtx.at(0));
+    }
+
+    /**
+     * Get the vout index of the segwit commitment in the given coinbase transaction.
+     *
+     * Returns -1 if no witness commitment was found.
+     */
+    template<typename T> static inline int GetWitnessCommitmentIndex(const T& coinbase) {
```

Comment:
Why is this moved? I think the goal for `primitives` is to be data-only and don't contain any helper logic, especially if they are consensus critical.

Reply:
It seems very weird to keep this obviously `CBlock`-related functionality in some completely different place. I'm not sure about primitives and data-only logic. If people feel strongly, I can restore this. Should the other stuff (witness commitment section stuff) follow along into validation.cpp then, you think?